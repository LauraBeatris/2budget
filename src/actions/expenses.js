import uuid from "uuid";

//ADD_EXPENSE
export const addExpense = ({
  description = "",
  note = "",
  amount = 0,
  createdAt = 0
} = {}) => {
  return {
    type: "ADD_EXPENSE",
    expense: {
      // Setting a id
      id: uuid(),
      description,
      note,
      amount,
      createdAt
    }
  };
};

//REMOVE_EXPENSE
export const removeExpense = ({ id } = {}) => ({
  type: "REMOVE_EXPENSE",
  id
});

//EDIT_EXPENSE
export const editExpense = (id, update) => ({
  type: "EDIT_EXPENSE",
  id,
  update
});
