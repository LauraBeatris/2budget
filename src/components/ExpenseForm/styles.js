import styled from "styled-components";

export const FormContainer = styled.div`
  display: flex;
  flex-direction: column;
  padding: 2rem;

  .msg-error {
    font-size: 1.5rem;
    color: rgb(209, 77, 53);
    margin-bottom: 1rem;
    font-weight: bold;
  }

  form {
    display: flex;
    flex-direction: column;

    justify-content: center;
    align-items: center;
  }

  .description-amount {
    padding: 1rem;
    margin-bottom: 2rem;
    text-align: center;

    input:first-child {
      margin-right: 1rem;
      border-radius: 8px;
      border: 1px solid rgb(77, 81, 94);
      box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
      color: rgb(77, 81, 94);
      font-style: italic;
      font-size: 1.5rem;
      padding: 0.5rem 1rem;
      text-transform: lowercase;
      font-style: italic;
      &::placeholder {
        font-weight: lighter;
        color: rgb(77, 81, 94);
      }
    }

    input:last-child {
      padding: 5px 10px;
      width: 100px;
      font-style: italic;
      font-weight: lighter;
      background-color: transparent;
      border: none;
      color: rgb(77, 81, 94);
      font-size: 1.8rem;
      border-bottom: 1px solid rgb(77, 81, 94);
    }
  }

  .note {
    border: 1px solid rgb(77, 81, 94, 0.9);
    background-color: #fff;
    overflow: auto;
    resize: none;
    padding: 2rem 2rem;
    border-radius: 15px;
    text-align: center;
    font-size: 2rem;
    color: #6dbfcc;
    margin-bottom: 2rem;

    &::placeholder {
      font-weight: bold;
      color: #6dbfcc;
      font-family: "Raleway";
    }
  }

  button.btn-submit {
    margin-top: 3rem;
    background-color: #6dbfcc;
    border: none;
    font-size: 1.6rem;
    font-weight: bold;
    padding: 2rem 4rem;
    color: white;
    border-radius: 100px;
    box-shadow: 0 1rem 2rem rgba(0, 0, 0, 0.2);
    cursor: pointer;
    text-decoration: none;
    transition: all 0.5s;

    &:hover {
      transform: translateY(-6px);
    }

    &:active {
      transform: translateY(0px);
      box-shadow: 0 2rem 2rem rgba(0, 0, 0, 0.2);
    }
  }
`;
