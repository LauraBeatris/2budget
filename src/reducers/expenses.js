const expensesReducerDefaultState = [];

const expensesReducer = (state = expensesReducerDefaultState, action) => {
  switch (action.type) {
    case "ADD_EXPENSE":
      return [...state, action.expense];
    case "REMOVE_EXPENSE":
      return state.filter(({ id }) => id !== action.id);
    case "EDIT_EXPENSE":
      return state.map(expense => {
        if (expense.id === action.id) {
          return {
            ...expense,
            ...action.update // overriden the expense items with the action updates icons
          };
        } else {
          return expense;
        }
      });
    default:
      return state;
  }
};

export default expensesReducer;
