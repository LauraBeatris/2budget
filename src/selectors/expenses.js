import moment from "moment";

// Get visible expenses
export const getVisibleExpenses = (
  expenses,
  { text, sortBy, startDate, endDate }
) => {
  return expenses
    .filter(expense => {
      const createdAtMoment = moment(expense.createdAt);
      const startDateMatch = startDate
        ? startDate.isSameOrBefore(createdAtMoment, "day")
        : true;
      const endDateMatch = endDate
        ? endDate.isSameOrAfter(createdAtMoment, "day")
        : true;

      const textMatch = expense.description
        .toLowerCase()
        .includes(text.toLowerCase());

      return startDateMatch && endDateMatch && textMatch;
    })
    .sort((a, b) => {
      if (sortBy === "date") {
        return b.createdAt < a.createdAt ? -1 : 1;
      } else if (sortBy === "amount") {
        return a.amount > b.amount ? -1 : 1;
      } else {
        return null;
      }
    });
};

export const getExpenseTotal = expenses => {
  if (expenses.length === 0) {
    return 0;
  } else if (expenses.length > 1) {
    const amounts = expenses.map(expense => expense.amount);
    return amounts.reduce((sum, value) => sum + value, 1);
  } else {
    return expenses[0].amount;
  }
};
