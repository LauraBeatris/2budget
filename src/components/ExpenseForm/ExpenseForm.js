import React, { Component } from "react";
import moment from "moment";
import { SingleDatePicker } from "react-dates";
import { FormContainer } from "./styles";

export default class ExpenseForm extends Component {
  constructor(props) {
    super(props);

    this.state = {
      description: props.expense ? props.expense.description : "",
      note: props.expense ? props.expense.note : "",
      amount: props.expense ? (props.expense.amount / 100).toString() : "",
      createdAt: props.expense ? moment(props.expense.createdAt) : moment(),
      calendarFocused: false,
      error: ""
    };
  }

  // Updating the description state with the input value
  onDescriptionChange = event => {
    const description = event.target.value;
    this.setState({ description });
  };

  // Updating the note state with the text area value
  onTextAreaChange = event => {
    const note = event.target.value;
    this.setState({ note });
  };

  // Updating the amount state with the text area value
  onAmountChange = event => {
    const amount = event.target.value;

    if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
      this.setState({ amount });
    }
  };

  // Handling the data changing and setting it to state
  onDateChange = createdAt => {
    // Prevent the user to clear the value
    if (createdAt) {
      this.setState({ createdAt });
    }
  };

  // Handling the focus changing and setting it to state
  onFocusChange = ({ focused }) => {
    this.setState({ calendarFocused: focused });
  };

  onSubmit = event => {
    event.preventDefault();

    // Checking if there's a description and a amount
    if (!this.state.description || !this.state.amount) {
      // Set error state equal to 'Please provide description and amount'
      const error = "Please provide description and amount";
      this.setState({ error });
    } else {
      // Clear the error
      this.setState({ error: "" });
      // Calling the onSubmit of from props to pass a expense
      this.props.onSubmit({
        description: this.state.description,
        amount: parseFloat(this.state.amount, 10) * 100,
        createdAt: this.state.createdAt.valueOf(),
        note: this.state.note
      });
    }
  };

  render() {
    return (
      <FormContainer>
        <form onSubmit={this.onSubmit}>
          {" "}
          <div>
            {this.state.error && (
              <p className="msg-error">{this.state.error}</p>
            )}
          </div>
          <div className="description-amount">
            <input
              type="text"
              placeholder="Type a description"
              value={this.state.description}
              onChange={this.onDescriptionChange}
              autoFocus
            />
            <input
              type="number"
              placeholder="amount"
              value={this.state.amount}
              onChange={this.onAmountChange}
            />
          </div>
          <textarea
            placeholder="Add a note for your expense (optional)"
            value={this.state.rent}
            onChange={this.onTextAreaChange}
            className="note"
            rows="2"
          />
          <SingleDatePicker
            date={this.state.createdAt}
            onDateChange={this.onDateChange}
            focused={this.state.calendarFocused}
            onFocusChange={this.onFocusChange}
            numberOfMonths={1}
            isOutsideRange={day => false}
          />
          <button className="btn-submit">{this.props.type} Expense</button>
        </form>
      </FormContainer>
    );
  }
}
