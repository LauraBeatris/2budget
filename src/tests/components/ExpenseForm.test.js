import React from "react";
import { shallow } from "enzyme";
import ExpenseForm from "../../components/ExpenseForm/ExpenseForm";
import expenses from "../fixtures/expenses";
import moment from "moment";

test("should render ExpenseForm correctly", () => {
  const wrapper = shallow(<ExpenseForm />);
  expect(wrapper).toMatchSnapshot();
});

test("should render ExpenseForm with expense data", () => {
  const wrapper = shallow(<ExpenseForm expense={expenses[0]} />);
  expect(wrapper).toMatchSnapshot();
});

test("should render error for invalid form submission", () => {
  const wrapper = shallow(<ExpenseForm />);
  // Before the changes
  expect(wrapper).toMatchSnapshot();

  // Finding the form and simulating the submit event
  wrapper.find("form").simulate("submit", { preventDefault: () => {} });

  // Checking if the error state changed
  expect(wrapper.state("error").length).toBeGreaterThan(0);

  // Making sure that the error always render correctly
  expect(wrapper).toMatchSnapshot();
});

test("should set description on input change", () => {
  const wrapper = shallow(<ExpenseForm />);
  const value = "Some fun description";
  // Matching the first input
  // Simulating the change event
  wrapper
    .find("input")
    .at(0)
    .simulate("change", {
      target: { value }
    });

  // Checking if the state changed
  expect(wrapper.state("description")).toBe(value);
});

// should set note on textarea change

test("should set note on textarea change", () => {
  const wrapper = shallow(<ExpenseForm />);
  const value = "Fun note";

  // Matching the element and simulating the change event
  wrapper.find("textarea").simulate("change", {
    target: {
      value
    }
  });

  // Checking if the state changed
  expect(wrapper.state("note")).toBe(value);
});

test("should set amount if valid input", () => {
  const wrapper = shallow(<ExpenseForm />);
  const value = "23.50";

  // Matching the element and simulating the change event
  wrapper
    .find("input")
    .at(1)
    .simulate("change", {
      target: {
        value
      }
    });

  // Checking if the state changed
  expect(wrapper.state("amount")).toBe(value);
});

test("should not set amount if invalid input", () => {
  const wrapper = shallow(<ExpenseForm />);
  const value = "20.202";

  // Matching the element and simulating the change event
  wrapper
    .find("input")
    .at(1)
    .simulate("change", {
      target: {
        value
      }
    });

  // Checking if the state changed
  expect(wrapper.state("amount")).toBe("");
});

test("should call onSubmit prop for valid form submission", () => {
  // Creating a new spy -> jest.fn() -> and storing on a variable
  const onSubmitSpy = jest.fn();
  const wrapper = shallow(
    <ExpenseForm onSubmit={onSubmitSpy} expense={expenses[0]} />
  );

  // Simulating the form submission
  wrapper.find("form").simulate("submit", {
    preventDefault: () => {}
  });

  // Checking if the error state was cleared
  expect(wrapper.state("error")).toBe("");

  // Checking if the spy was called with the specific args
  expect(onSubmitSpy).toHaveBeenLastCalledWith({
    description: expenses[0].description,
    note: expenses[0].note,
    amount: expenses[0].amount,
    createdAt: expenses[0].createdAt.valueOf()
  });
});

test("should set new date on date changes", () => {
  const wrapper = shallow(<ExpenseForm />);
  const now = moment();

  // Matching the element
  // Accessing props to get the handler and calling it with a moment instance
  wrapper.find("SingleDatePicker").prop("onDateChange")(now);

  // Checking the state
  expect(wrapper.state("createdAt")).toBe(now);
});

test("should set calendar focus on change", () => {
  const wrapper = shallow(<ExpenseForm />);
  const focused = true;

  // Matching the element
  // Accessing props to get the handler and calling it with a focused object
  wrapper.find("SingleDatePicker").prop("onFocusChange")({ focused });

  // Checking the state
  expect(wrapper.state("calendarFocused")).toBe(focused);
});
