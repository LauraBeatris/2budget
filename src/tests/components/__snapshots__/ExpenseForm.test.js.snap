// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseForm correctly 1`] = `
<.sc-bdVaJa>
  <form
    onSubmit={[Function]}
  >
     
    <div />
    <div
      className="description-amount"
    >
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="Type a description"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="amount"
        type="number"
        value=""
      />
    </div>
    <textarea
      className="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      rows="2"
    />
    <SingleDatePicker
      anchorDirection="left"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month",
          "jumpToPrevMonth": "Move backward to switch to the previous month",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
          "moveFocusByOneMonth": "Switch months",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
          "openThisPanel": "Open this panel",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field",
          "selectFocusedDate": "Select the date in focus",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <button
      className="btn-submit"
    >
       Expense
    </button>
  </form>
</.sc-bdVaJa>
`;

exports[`should render ExpenseForm with expense data 1`] = `
<.sc-bdVaJa>
  <form
    onSubmit={[Function]}
  >
     
    <div />
    <div
      className="description-amount"
    >
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="Type a description"
        type="text"
        value="Water rent"
      />
      <input
        onChange={[Function]}
        placeholder="amount"
        type="number"
        value="1.95"
      />
    </div>
    <textarea
      className="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      rows="2"
    />
    <SingleDatePicker
      anchorDirection="left"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month",
          "jumpToPrevMonth": "Move backward to switch to the previous month",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
          "moveFocusByOneMonth": "Switch months",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
          "openThisPanel": "Open this panel",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field",
          "selectFocusedDate": "Select the date in focus",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <button
      className="btn-submit"
    >
       Expense
    </button>
  </form>
</.sc-bdVaJa>
`;

exports[`should render error for invalid form submission 1`] = `
<.sc-bdVaJa>
  <form
    onSubmit={[Function]}
  >
     
    <div />
    <div
      className="description-amount"
    >
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="Type a description"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="amount"
        type="number"
        value=""
      />
    </div>
    <textarea
      className="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      rows="2"
    />
    <SingleDatePicker
      anchorDirection="left"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month",
          "jumpToPrevMonth": "Move backward to switch to the previous month",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
          "moveFocusByOneMonth": "Switch months",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
          "openThisPanel": "Open this panel",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field",
          "selectFocusedDate": "Select the date in focus",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <button
      className="btn-submit"
    >
       Expense
    </button>
  </form>
</.sc-bdVaJa>
`;

exports[`should render error for invalid form submission 2`] = `
<.sc-bdVaJa>
  <form
    onSubmit={[Function]}
  >
     
    <div>
      <p
        className="msg-error"
      >
        Please provide description and amount
      </p>
    </div>
    <div
      className="description-amount"
    >
      <input
        autoFocus={true}
        onChange={[Function]}
        placeholder="Type a description"
        type="text"
        value=""
      />
      <input
        onChange={[Function]}
        placeholder="amount"
        type="number"
        value=""
      />
    </div>
    <textarea
      className="note"
      onChange={[Function]}
      placeholder="Add a note for your expense (optional)"
      rows="2"
    />
    <SingleDatePicker
      anchorDirection="left"
      customCloseIcon={null}
      customInputIcon={null}
      date={"1970-01-01T00:00:00.000Z"}
      daySize={39}
      disabled={false}
      displayFormat={[Function]}
      enableOutsideDays={false}
      firstDayOfWeek={null}
      focused={false}
      hideKeyboardShortcutsPanel={false}
      horizontalMargin={0}
      id="date"
      initialVisibleMonth={null}
      inputIconPosition="before"
      isDayBlocked={[Function]}
      isDayHighlighted={[Function]}
      isOutsideRange={[Function]}
      isRTL={false}
      keepOpenOnDateSelect={false}
      monthFormat="MMMM YYYY"
      navNext={null}
      navPrev={null}
      numberOfMonths={1}
      onClose={[Function]}
      onDateChange={[Function]}
      onFocusChange={[Function]}
      onNextMonthClick={[Function]}
      onPrevMonthClick={[Function]}
      openDirection="down"
      orientation="horizontal"
      phrases={
        Object {
          "calendarLabel": "Calendar",
          "chooseAvailableDate": [Function],
          "clearDate": "Clear Date",
          "closeDatePicker": "Close",
          "dateIsUnavailable": [Function],
          "enterKey": "Enter key",
          "escape": "Escape key",
          "hideKeyboardShortcutsPanel": "Close the shortcuts panel",
          "homeEnd": "Home and end keys",
          "jumpToNextMonth": "Move forward to switch to the next month",
          "jumpToPrevMonth": "Move backward to switch to the previous month",
          "keyboardNavigationInstructions": "Press the down arrow key to interact with the calendar and
  select a date. Press the question mark key to get the keyboard shortcuts for changing dates.",
          "keyboardShortcuts": "Keyboard Shortcuts",
          "leftArrowRightArrow": "Right and left arrow keys",
          "moveFocusByOneDay": "Move backward (left) and forward (right) by one day",
          "moveFocusByOneMonth": "Switch months",
          "moveFocusByOneWeek": "Move backward (up) and forward (down) by one week",
          "moveFocustoStartAndEndOfWeek": "Go to the first or last day of a week",
          "openThisPanel": "Open this panel",
          "pageUpPageDown": "page up and page down keys",
          "questionMark": "Question mark",
          "returnFocusToInput": "Return to the date input field",
          "selectFocusedDate": "Select the date in focus",
          "showKeyboardShortcutsPanel": "Open the keyboard shortcuts panel",
          "upArrowDownArrow": "up and down arrow keys",
        }
      }
      placeholder="Date"
      readOnly={false}
      renderCalendarInfo={null}
      renderDay={null}
      renderMonth={null}
      reopenPickerOnClearDate={false}
      required={false}
      screenReaderInputMessage=""
      showClearDate={false}
      showDefaultInputIcon={false}
      weekDayFormat="dd"
      withFullScreenPortal={false}
      withPortal={false}
    />
    <button
      className="btn-submit"
    >
       Expense
    </button>
  </form>
</.sc-bdVaJa>
`;
